ADAL/src/cache/ADTokenCache.m

```diff
@@ -69,6 +69,45 @@ - (void)dealloc
     SAFE_ARC_SUPER_DEALLOC();
 }
 
+- (id)cacheObjectForKey:(id<NSCopying>)key
+{
+    id object = nil;
+    @synchronized(self)
```

Comment:
The usage pattern for ADAL cache tends to be fairly read heavy, as such I would like to avoid mutexes and these small methods.